compiler/rustc_session/src/session.rs

```diff
@@ -1512,6 +1512,17 @@ fn validate_commandline_args_with_session_available(sess: &Session) {
     if let (Some(first), Some(second)) = (sanitizer_iter.next(), sanitizer_iter.next()) {
         sess.err(&format!("`-Zsanitizer={}` is incompatible with `-Zsanitizer={}`", first, second));
     }
+
+    // Cannot enable crt-static with sanitizers on Linux
+    for sanitizer in sess.opts.debugging_opts.sanitizer.into_iter() {
```

Comment:
Instead of emitting an error message for each of the enabled sanitizers, could we emit a single message that covered all of the enabled sanitizers in a single diagnostic?